### DISCUSSION TITLE
JSON Integration

### DISCUSSION THREAD

**Discussion by BrandonUSapp** [aws/graph-explorer#204] [Q&A]
Is there a way to launch a Graph Explorer instance using a JSON as a data source instead of a live connection to Neptune? Or even slow load with DynamoDB?

**Comment by triggan** [comment]
It's possible, but likely a non-trivial set of changes would need to be made to support thius.  Graph Explorer currently generates visualizations through a two-step process.  First it derives a graph schema through either a set of queries or (if available) Neptune's Summary API [1].  That schema is then used to allow a user to select/query nodes and edges to visualize.  When a user wants to expand the graph or query for specific items, additional queries are made to the backend database to fetch those required objects.  Today, all of the querying is accomplished via one of the supported graph query languages (Gremlin, openCypher, or SPARQL).  The implementation is abstract enough such that many other graph databases that support the same graph query languages could be used (not just Neptune).

To support this with other, non-graph backends, it would require some other means of supplying a graph schema / mapping to the underlying datastore/format.  It would also require building the data querying/fetching capabilities that tie into the front-end functions for querying and expanding the graph.

[1] https://docs.aws.amazon.com/neptune/latest/userguide/neptune-graph-summary.html
